packages/expo-dev-launcher/bundle/DevMenu.ts

```diff
@@ -18,8 +18,8 @@ export async function queryMyProjectsAsync(): Promise<any> {
   return await DevMenu.queryMyProjectsAsync();
 }
 
-export async function queryDevSessionsAsync(): Promise<any> {
-  return await DevMenu.queryDevSessionsAsync();
+export async function queryDevSessionsAsync(deviceID?: string): Promise<any> {
+  return await DevMenu.queryDevSessionsAsync(deviceID ?? null);
```

Comment:
if this is a fetch call, could we not move it into JS in the future? e.g `DevMenu.getDeviceId()` or even just use the `initialProp` and reuse the same logic for both platforms?

Reply:
My understanding is that we are trying to keep auth secrets in the dev menu's native code, so while we could implement the `deviceID` query in JS, we'd still have to keep the authenticated dev session request in the native module -- so we'd essentially be duplicating logic.

Reply:
ah okay - I'd love to have a better understanding of why this is the case @lukmccall can you clarify?